### DISCUSSION TITLE
Kafka pods not deployed

### DISCUSSION THREAD

**Discussion by Muddam234** [strimzi/strimzi-kafka-operator#9026] [General]
Hi All,

Recently i have created Kubernetes cluster and then i have deployed Strimzi CRDs 0.23 and all crds successfully deployed and then i have deployed Kafka and i see my deployment is created but when i check the pods its not showing in the cluster. and i am not able to find the logs as well. Can any one please help on this issue.

deployment status:
\kafka\gcp-qa>kubectl apply -f kafka-gcp.yaml -n kafka-testing-poc
kafka.kafka.strimzi.io/kafka-testing-poc-gcp created
configmap/kafka-metrics created

Strimzi cluster operator logs:

2023-08-24 14:21:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:23:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:25:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:27:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:29:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:31:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:33:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...
2023-08-24 14:35:13 INFO  ClusterOperator:125 - Triggering periodic reconciliation for namespace strimzi...

**Comment by scholzj** [comment]
You should be really using Strimzi 0.23 these days. It is several years old. You should use some new version.

Also, from what you shared, you are using wrong namespaces. You installed Strimzi to watch namespace `strimzi` but you create your custom resource in namespace `kafka-testing-poc`.

**Reply by Muddam234** [reply]
but Strimzi cluster operator running in strimzi namespace and kafka deployed in kafka-testing-poc namespace and my production cluster also running same way. so am replicating same as my production for migrating 0.23 to latest version.

**Reply by scholzj** [reply]
But you probably configured it differently there. It can run in a different namespace than where the Kafka cluster is, but you have to configure what namespace(s) to watch. In your case, as the log says, it is configured to watch the namespace `strimzi`.

**Reply by Muddam234** [reply]
ok let me reverify my configuration and get back to same and thanks Scholzj

**Reply by Muddam234** [reply]
Sorry for later respond, now i have deployed with same namespace and after that my zookeeper is up and running and kafka is pod not showing and when i check the PVC its in pending status in kafka PVC volumes.

Error: 
│ Events:                                                                                                                           │
│   Type    Reason                Age                        From                         Message                                   │
│   ----    ------                ----                       ----                         -------                                   │
│   Normal  WaitForFirstConsumer  4m29s (x17025 over 2d22h)  persistentvolume-controller  waiting for first consumer to be created  │
│ before binding

logs in Strimzi-cluster-operator: 

 2023-08-28 10:06:38 ERROR AbstractOperator:276 - Reconciliation #2136(timer) Kafka(strimzi/strimzi-gcp): createOrUpdate failed    │
│ io.strimzi.operator.common.operator.resource.TimeoutException: Exceeded timeout of 300000ms while waiting for Service resource st │
│ rimzi-gcp-kafka-externalint-bootstrap in namespace strimzi to be addressable                                                      │
│     at io.strimzi.operator.common.Util$1.lambda$handle$1(Util.java:137) ~[io.strimzi.operator-common-0.23.0.jar:0.23.0]           │
│     at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:128) ~[io.vertx.vertx-core-4.0.3.jar:4.0.3]               │
│     at io.vertx.core.impl.future.FutureBase.lambda$emitFailure$1(FutureBase.java:71) ~[io.vertx.vertx-core-4.0.3.jar:4.0.3]       │
│     at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [io.netty.netty-common-4.1.60.F │
│ inal.jar:4.1.60.Final]                                                                                                            │
│     at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [io.netty.netty-common- │
│ 4.1.60.Final.jar:4.1.60.Final]                                                                                                    │
│     at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) [io.netty.netty-transport-4.1.60.Final.jar:4.1.60.Final]      │
│     at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [io.netty.netty-common-4.1.60 │
│ .Final.jar:4.1.60.Final]                                                                                                          │
│     at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.F │
│ inal]                                                                                                                             │
│     at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.60.Final. │
│ jar:4.1.60.Final]                                                                                                                 │
│     at java.lang.Thread.run(Thread.java:829) [?:?]                                                                                │
│ 2023-08-28 10:06:38 WARN  AbstractOperator:516 - Reconciliation #2136(timer) Kafka(strimzi/strimzi-gcp): Failed to reconcile

**Reply by scholzj** [reply]
This looks like a different error. You need to start by sharing your full `Kafka` CR and full logs from the operator.

**Reply by Muddam234** [reply]
Cluster-Operator-Logs: 
[cluster-operator-pod.log](https://github.com/strimzi/strimzi-kafka-operator/files/12454760/cluster-operator-pod.log)

**Reply by Muddam234** [reply]
@scholzj  I saw the error can you please help .

events logs:
│     at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.F │
│     at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.60.Final. │
│     at java.lang.Thread.run(Thread.java:829) [?:?]                                                                                │
│ 2023-09-04 07:16:21 WARN  AbstractOperator:516 - Reconciliation #3508(timer) Kafka(strimzi/strimzi-gcp): Failed to reconcile      │
│ io.strimzi.operator.common.operator.resource.TimeoutException: Exceeded timeout of 300000ms while waiting for Ingress resource st │
│     at io.strimzi.operator.common.Util$1.lambda$handle$1(Util.java:137) ~[io.strimzi.operator-common-0.23.0.jar:0.23.0]           │
│     at io.vertx.core.impl.future.FutureImpl$3.onFailure(FutureImpl.java:128) ~[io.vertx.vertx-core-4.0.3.jar:4.0.3]               │
│     at io.vertx.core.impl.future.FutureBase.lambda$emitFailure$1(FutureBase.java:71) ~[io.vertx.vertx-core-4.0.3.jar:4.0.3]       │
│     at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [io.netty.netty-common-4.1.60.F │
│     at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [io.netty.netty-common- │
│     at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) [io.netty.netty-transport-4.1.60.Final.jar:4.1.60.Final]      │
│     at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [io.netty.netty-common-4.1.60 │
│     at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty.netty-common-4.1.60.Final.jar:4.1.60.F │
│     at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty.netty-common-4.1.60.Final. │
│     at java.lang.Thread.run(Thread.java:829) [?:?]


Error syncing to GCP: error running backend syncing routine: googleapi: Error 404: The resource 'projects/qa-testing-kafka-56417/zones/us-central1-a/networkEndpointGroups/k8s1-c9cc3225-strimz-strimzi-gcp-kafka-externalint-909-894c38bb' was not found, notFound
